### DISCUSSION TITLE
Why does custom script logging not print objects?

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#17974] [Q&A]
For debugging custom scripts, sometimes I like to include a line like this:

```python
self.log_debug(f"User entered data: {data}")
```

But the Netbox `/extras/scripts/results/xxx/` page does not print the object names:
> User entered data: {'site': , 'location': , 'rack': None, 'rack_position': None, 'rack_face': '', 'manufacturer': None, 'device_type': , 'role': , 'device_description': 'deleteme', 'status': 'active', 'serial': '', 'location_served': None, 'primary_nic': , 'primary_mac': '', 'primary_ip': None, 'ip_range': None, 'vrf': None, 'switch': None, 'switch_interface': None, 'pdu': None, 'pdu_outlet': None}

If I configure Netbox to send log output to `logging.handlers.SysLogHandler` then I can see the full output like this:
> User entered data: {'site': <Site: ABQ-ASA>, 'location': <Location: 102>, 'rack': None, 'rack_position': None, 'rack_face': '', 'manufacturer': None, 'device_type': <DeviceType: Kumo CP>, 'role': <DeviceRole: AV>, 'device_description': 'deleteme', 'status': 'active', 'serial': '', 'location_served': None, 'primary_nic': <InterfaceTemplate: GigabitEthernet1/0/1>, 'primary_mac': '', 'primary_ip': None, 'ip_range': None, 'vrf': None, 'switch': None, 'switch_interface': None, 'pdu': None, 'pdu_outlet': None}

Is there an easy way to make the Netbox UI output look the same?

**Comment by llamafilm** [comment]
Aha!  I answered my own question.  Looks like this is some sort of HTML tag sanitization.  It prints correctly with backticks:

```python
self.log_debug(f"User entered data: `{data}`")
```

**Reply by candlerb** [reply]
The text from `log` is processed as Markdown, and in Markdown HTML tags are passed through as HTML, so you could say it's really a *lack* of HTML tag sanitization.
